ext/standard/streamsfuncs.c

```diff
@@ -616,93 +616,147 @@ PHP_FUNCTION(stream_get_wrappers)
 }
 /* }}} */
 
-/* {{{ stream_select related functions */
-static int stream_array_to_fd_set(zval *stream_array, fd_set *fds, php_socket_t *max_fd)
-{
-	zval *elem;
-	php_stream *stream;
-	int cnt = 0;
+/* {{{ fd_bigset related macros and typedefs to remove FD_SETSIZE limitation in stream_select */
+typedef struct {
+    char *fds_bits;
+    size_t size;
+} fd_bigset;
+#define FD_BIGSET_ENSURE_CAPACITY(fd, set) \
+    if (((fd) / 8 >= (set)->size)) { \
+        fd_bigset_double_size(set); \
+    }
+#define FD_BIGSET_ZERO(set, num_fds) do { \
+    (set)->size = (num_fds + 7) / 8; \
+    (set)->fds_bits = (char *)ecalloc((set)->size, sizeof(char)); \
+} while (0)
+#define FD_BIGSET_SET(fd, set) do { \
+	FD_BIGSET_ENSURE_CAPACITY(fd, set) \
+    if ((fd) / 8 < (set)->size) { \
+        (set)->fds_bits[(fd) / 8] |= (1 << ((fd) % 8)); \
+    } \
+} while (0)
+#define FD_BIGSET_ISSET(fd, set) (((fd) / 8 < (set)->size) ? ((set)->fds_bits[(fd) / 8] & (1 << ((fd) % 8))) : 0)
+#define FD_BIGSET_CLR(fd, set) do { \
+    if ((fd) / 8 < (set)->size) { \
+        (set)->fds_bits[(fd) / 8] &= ~(1 << ((fd) % 8)); \
+    } \
+} while (0)
+#define FD_BIGSET_FREE(set) do { \
+    if ((set)->fds_bits) { \
+        efree((set)->fds_bits); \
+        (set)->fds_bits = NULL; \
+    } \
+    (set)->size = 0; \
+} while (0)
+
+/* {{{ Function to double the size of an fd_bigset */
+void fd_bigset_double_size(fd_bigset *set) {
+    if (!set || !set->fds_bits) {
+        return;
+    }
 
-	if (Z_TYPE_P(stream_array) != IS_ARRAY) {
-		return 0;
-	}
+    size_t old_size = set->size;
+    size_t new_size = old_size * 2;
+
+    /* Allocate new memory block twice the size of the original */
+    char *new_bits = (char *)ecalloc(new_size, sizeof(char));
+
+    /* Copy old bits to new memory block */
+    memcpy(new_bits, set->fds_bits, old_size);
+
+    /* Free the old memory block */
+    efree(set->fds_bits);
```

Comment:
You can use `set->fds_bits = erealloc(set->fds_bits, new_size)` here (you just need a memset() to zero out the extra capacity)